src/cpp/core/include/core/FileInfo.hpp

```diff
@@ -89,7 +89,7 @@ class FileInfo
    }
    
 public:
-   std::string absolutePath() const { return absolutePath_.c_str(); }
+   const std::string& absolutePath() const { return absolutePath_; }
```

Comment:
Why this change? Isn't it generally risky to return a reference to a class member? Since you then just do `c_str()`, anyways, why not keep it the same? Performance?

Reply:
Right, the old implementation was forcing a copy every time we requested the absolute path of a FileInfo object:

- We call `.c_str()`, returning a `const char*`,
- That's implicitly converted to `std::string`, constructing a new `std::string` from that C string,
- That also implies finding the null terminator when constructing that new `std::string`.

Returning a const-ref is risky, since it implies someone could store a reference that outlives the parent object. I'll change this back to return a plain `std::string` -- at least then we're still just creating a `std::string` from a `std::string` and the compiler is likely to optimize any copies away anyhow.